### PR TITLE
fix(style): design fixes for table

### PR DESCRIPTION
**Pull request by sehyunidaaa**
Minor design updates for table:

- Header row text is left-aligned
- Header row text has smaller font size

### CODE REVIEW

**Comment by seaerchin** [comment]
!run chromatic

File: packages/components/src/templates/next/components/native/Table/Table.tsx [hunk 1]
```diff
@@ -78,18 +78,18 @@ const Table = ({ caption, content }: TableProps) => {
...
                     <TableCellTag
                       key={cellIndex}
                       colSpan={cell.colSpan}
                       rowSpan={cell.rowSpan}
-                      className={`border-divide-subtle max-w-40 break-words border-b border-r px-4 py-3.5 align-top first:border-l last:max-w-full [&_li]:my-0 [&_li]:pl-1 [&_ol]:mt-0 [&_ol]:ps-5 [&_ol]:text-sm [&_ul]:mt-0 [&_ul]:ps-5 ${
+                      className={`border-divide-subtle max-w-40 break-words border-b border-r px-4 py-3.5 align-top first:border-l last:max-w-full [&_li]:my-0 [&_li]:pl-1 [&_ol]:mt-0 [&_ol]:ps-5 [&_ol]:text-sm [&_p]:text-sm [&_ul]:mt-0 [&_ul]:ps-5 ${
                         cell.type === "tableHeader"
                           ? "bg-utility-neutral"
-                          : "bg-utility-neutral-subtle [&_p]:text-sm"
+                          : "bg-utility-neutral-subtle"
```
Comment (seaerchin) [inline packages/components/src/templates/next/components/native/Table/Table.tsx:92]: actually just a point here - we now rely exclusively on the background to differentiate between header-row and cell. 

going off notion's one, they seem to have the header row having slightly heavier `font-weight` (and is also what i'd expect off my web browsing habits). wondering how much extra effort is required for this to conform w/ user's expectations
image

**Review by seaerchin — APPROVED** [review]
lgtm, 1 minor comment but not a blocker

File: packages/components/src/templates/next/components/native/Table/Table.tsx (hunk 1 above)
Comment (sehyunidaaa) [reply packages/components/src/templates/next/components/native/Table/Table.tsx:92]: using font-weight to differentiate header as well! this screenshot is from this branch:

image

**Pull request merged**